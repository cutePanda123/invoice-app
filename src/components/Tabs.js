import React from 'react';
import PropTypes from 'prop-types';

export class Tabs extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            activeIndex: this.props.activeIndex
        };
    }

    clickTab = (event, index) => {
        event.preventDefault();
        this.setState({
            activeIndex: index
        });
        this.props.onTabChange(index);
    }

    render() {
        const { children } = this.props;
        const { activeIndex } = this.state;
        return (
            <ul className="nav nav-tabs nav-fill my-4">
                {
                    React.Children.map(children, (child, index) => {
                        const curClassName = activeIndex === index ? 'nav-link active' : 'nav-link';
                        return (
                            <li className="nav-item">
                                <a
                                    onClick={(event) => {this.clickTab(event, index);}}
                                    className={curClassName}
                                    href="#"
                                >
                                    {child}
                                </a>
                            </li>
                        );
                    })
                }
            </ul>
        );
    }
};

Tabs.propTypes = {
    activeIndex: PropTypes.number.isRequired,
    onTabChange: PropTypes.func.isRequired
};

export const Tab = ({children}) => {
    return <React.Fragment>{children}</React.Fragment>;
};