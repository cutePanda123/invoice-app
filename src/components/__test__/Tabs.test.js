import React from 'react';
import { shallow } from 'enzyme';
import { Tabs, Tab } from '../Tabs';
import Utility from '../../utility';

const props = {
    activeIndex: 0,
    onTabChange: jest.fn()
}

let wrapper = null;

describe('Tabs test', () => {
    beforeEach(() => {
        wrapper = shallow(
        <Tabs {...props}>
                <Tab>tab1</Tab>
                <Tab>tab2</Tab>
            </Tabs>);
    });

    it('snapshot test', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render correct number of tabs', () => {
        const tabs = wrapper.find(Tab);
        expect(tabs.length).toEqual(2);
    });

    it('should trigger correct callback on list view', () => {
        const tab1 = wrapper.find('.nav-item a').first();
        tab1.simulate('click', { preventDefault: () => {} });
        expect(props.onTabChange).toHaveBeenCalledWith(0);
    });

    it('should trigger correct callback on chart view', () => {
        const tab = wrapper.find('.nav-item a').last();
        tab.simulate('click', { preventDefault: () => {} });
        expect(props.onTabChange).toHaveBeenCalledWith(1);
    });
});