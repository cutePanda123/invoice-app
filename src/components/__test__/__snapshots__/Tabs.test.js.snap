// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs test snapshot test 1`] = `
<ul
  className="nav nav-tabs nav-fill my-4"
>
  <li
    className="nav-item"
    key=".0"
  >
    <a
      className="nav-link active"
      href="#"
      onClick={[Function]}
    >
      <Tab>
        tab1
      </Tab>
    </a>
  </li>
  <li
    className="nav-item"
    key=".1"
  >
    <a
      className="nav-link"
      href="#"
      onClick={[Function]}
    >
      <Tab>
        tab2
      </Tab>
    </a>
  </li>
</ul>
`;
